client/state/reader/follows/reducer.js

```diff
@@ -130,7 +133,18 @@ export const items = createReducer( {}, {
 	[ READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL ]: ( state, action ) => updatePostSubscription( state, action ),
 	[ READER_SUBSCRIBE_TO_NEW_COMMENT_EMAIL ]: ( state, action ) => updatePostSubscription( state, action ),
 	[ READER_UNSUBSCRIBE_TO_NEW_COMMENT_EMAIL ]: ( state, action ) => updatePostSubscription( state, action ),
-} );
+	[ READER_FOLLOWS_SYNC_COMPLETE ]: ( state, action ) => {
+		const seenSubscriptions = new Set( action.payload );
+
+		// diff what we saw vs. what's in state and remove anything extra
+		// extra would be active subscriptions that were not seen in the sync
+		return omitBy( state, ( follow ) => follow.ID &&
```

Comment:
after some offline discussion:
1. lets remove the `follow.is_following`
2. add in a comment saying that: "we need to check for a follow.ID because ..."